src/GlobalScreen/Scope/Layout/MetaContent/Media/CssCollection.php

```diff
@@ -13,12 +13,6 @@ class CssCollection extends AbstractCollection
      */
     public function addItem(Css $item)
     {
-        $basename = basename(parse_url($item->getContent(), PHP_URL_PATH));
-        foreach ($this->getItems() as $css) {
-            if (basename(parse_url($css->getContent(), PHP_URL_PATH)) === $basename) {
-                return;
-            }
-        }
```

Comment:
@Amstutz we have to find another solution here, we can't record all of them, because certain css may be added dozens of times (unfortunately i can't do much about that). using only the base name is obviously not enough, a hash from the whole path would be better. 

/templates/default/delos.css
and 
./templates/default/delos.css

as two different ones.
However, a complicated disassembly of the path at this point may well have an influence on the performance.

I'll have to think about it...

Translated with www.DeepL.com/Translator (free version)

Reply:
@chfsx thanks a lot for the quick feedback!

how and why are they added dozens of times? can you give some examples? I did not look far but on the few screens I checked we are fine. I agree, this can not be your responsibility to prevent that. This is something in need of fixing outside the global screen IMO.

Normalizing the path here should be quite cheap. We could do that. However, I would rather like to find the places were we load css multiple times and get rid of those. This seems like an issue to me that we just cover up like that.

If you say those lines are required (for the moment), I can re-add them. The delos.css blocking files includes from calendar are gone with this PR. But note, this crashes again as soon as somebody tries to add SOMETHING/delos.css again before delos.css is loaded. So this is a ticking time bomb.

Reply:
yes then we leave out the lines for the moment, with CSS the problem is much smaller than with JS-files anyway, in many places it is "ensured" that a certain JS-library is available.

Reply:
Why not using realpath() and comparing the results, this should be pretty safe.